app/screens/thread/thread.tsx

```diff
@@ -20,14 +20,15 @@ import EphemeralStore from '@store/ephemeral_store';
 
 import ThreadPostList from './thread_post_list';
 
+import type {CallsChannelState} from '@calls/observers';
 import type PostModel from '@typings/database/models/servers/post';
 import type {AvailableScreens} from '@typings/screens/navigation';
 import type {KeyboardTrackingViewRef} from 'react-native-keyboard-tracking-view';
 
 type ThreadProps = {
     componentId: AvailableScreens;
     isCRTEnabled: boolean;
-    isInACall: boolean;
+    callsChannelState: CallsChannelState;
```

Comment:
> how is passing the entire state impacting re-renders?
>
> Could it be better to always mount a component that checks for this values and renders null in the case is needed?

Unfortunately we do need these three booleans in the thread and channel screens, to determine whether to show the calls components, and where the place the `MoreMessages` banner. As for always mounting a component, we were doing that in the past but you suggested (rightly, I think) that we should not mount it if it's not needed.

So I think we're left with the question: should the three booleans be 1) wrapped in an `Observable<object>`, a 2) Object of Observables (which could be three-dot spread into the props object), or 3) individual `Observable<boolean>`s.

1 and 2 let us reduce code duplication. I tried both and chose 1 in https://github.com/mattermost/mattermost-mobile/commit/3d5a27b60ba567e3171e1a1c7337e0928af20010 . 3 means we have to duplicate some code, but I think it's clearer what's happening, and there's a chance that, because the booleans are directly observed, that rxjs/react might be able to do some value comparisons and cut out a rerender (where as 1, because it's an object, might cause a rerender because the object is different, even though only one of the the booleans changed).
 
So now I'm using 3 in https://github.com/mattermost/mattermost-mobile/pull/7283/commits/8ec56196b28f5597c86b6346cf6aa4173a22797b -- although there's duplication, I think is the best we can do to limit rerenders due to calls code.  

Incidentally, I did some debugging and it looks like calls only causes rerenders when one of those booleans changes, which is as expected and necessary. But `PostList` is being called multiple times (like, dozens of times) in longer threads -- maybe we should look at that, because it's pretty slow to render on my old iPhone 7 plus.